WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+
+
+
+// MARK: - ManagedObject Base Class
+//
+protocol ManagedObject
+{
+    static var entityName: String { get }
+}
+
+
+// MARK: - Core Data Helpers
+//
+struct CoreDataHelper<T where T: NSManagedObject, T: ManagedObject>
+{
+    /// CoreData ManagedObjectContext
+    ///
+    let context: NSManagedObjectContext
+
+
+    /// Returns a new FetchRequest for the associated Entity
+    ///
+    func newFetchRequest() -> NSFetchRequest {
+        return NSFetchRequest(entityName: T.entityName)
+    }
+
+
+    /// Returns all of the entities that match with a given predicate.
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
+    ///
+    func allObjects(matchingPredicate predicate: NSPredicate? = nil, sortedBy descriptors: [NSSortDescriptor]? = nil)  -> [T] {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.sortDescriptors = descriptors
+
+        return loadObjects(withFetchRequest: request)
+    }
+
+    /// Returns the number of entities found that match with a given predicate.
+    ///
+    /// - Parameter predicate: Defines the conditions that any given object should meet. Optional.
+    ///
+    func countObjects(matchingPredicate predicate: NSPredicate? = nil) -> Int {
+        let request = newFetchRequest()
+        request.predicate = predicate
+        request.includesSubentities = false
+        request.predicate = predicate
+
+        var result = 0
+
+        do {
+            result = try context.countForFetchRequest(request)
+        } catch {
+            DDLogSwift.logError("Error counting objects [\(T.entityName)]")
+        }
+
+        return result
+    }
+
+    /// Deletes the specified Object Instance
+    ///
+    func deleteObject(object: T) {
+        do {
+            context.deleteObject(object)
```

Comment:
1. How do you delete an object with this class without saving the changes in the context?  ie: temporary contexts
2. How do you use this method together with `NSManagedObjectContext`'s [perform method](https://developer.apple.com/reference/coredata/nsmanagedobjectcontext/1506578-perform)?